### PR TITLE
Add support for nlmsg extended permission

### PR DESCRIPTION
**Pull request by tweksteen**

### CODE REVIEW

**Comment by pebenito** [comment]
you'll need to at least update th github action to use main, so the builds will work. Also you should look at the tests and add some nlmsg examples to prove that this works.

**Comment by tweksteen** [comment]
Thanks for the review.

I changed the default SELinux branch in the github action as you suggested. I assume this implies that setools will require SELinux >= 3.8 in the future. Is that ok?

There is also a commit to rename IoctlSet to XpermSet. I don't know the compatibility guarantee for the setools API, but its worth mentioning. (This commit is not strictly necessary, but it helps clarifying the role of the class).

I added a test in `test_terulequery.py`. I don't know what's the best place otherwise. Please let me know and I can relocate it or add further tests. Also, I came across the issue that the tests in `TERuleQueryXperm` weren't being executed (as the test class does not starts with "Test"). This is fixed in the commit as well.

**Comment by pebenito** [comment]
> Thanks for the review.
> 
> I changed the default SELinux branch in the github action as you suggested. I assume this implies that setools will require SELinux >= 3.8 in the future. Is that ok?

That's fine. SETools needs to keep up with new SELinux features.


> There is also a commit to rename IoctlSet to XpermSet. I don't know the compatibility guarantee for the setools API, but its worth mentioning. (This commit is not strictly necessary, but it helps clarifying the role of the class).

SETools only has best-effort compat guarantees. This rename should be ok, but we should probably have a `IoctlSet` class that inherits `XpermSet` and throws a deprecation warning in its `__init__` before calling `super().__init__`.


> I added a test in `test_terulequery.py`. I don't know what's the best place otherwise. Please let me know and I can relocate it or add further tests.

I'd add one test in `tests/library/policyrep/test_rules.py`.

Then in `tests/library/policyrep/selinuxpolicyy.conf` replace one of each of the xperm rule types (allowxperm, auditallowxperm, etc.) to make sure this counts towards the rule counts. Don't add rules, as I intentionally made the rule counts prime numbers.


> Also, I came across the issue that the tests in `TERuleQueryXperm` weren't being executed (as the test class does not starts with "Test"). This is fixed in the commit as well.

Thanks!

**Comment by pebenito** [comment]
Looks like your build issues are due to issues I've since fixed in #139. You should also rebase to get those fixes.

**Comment by tweksteen** [comment]
Done. Let me know your thoughts. Thanks.

File: setools/policyrep/terule.pxi [hunk 1]
```diff
@@ -249,7 +249,7 @@ cdef class IoctlSet(frozenset):
         elif spec == ",":
             return ", ".join(shortlist)
         else:
-            return super(IoctlSet, self).__format__(spec)
+            return super(XpermSet, self).__format__(spec)
```
Comment (pebenito) [inline setools/policyrep/terule.pxi:252]: PR looks good, one last nit. `return super().__format__(spec)

**Review by pebenito — CHANGES_REQUESTED** [review]

File: setools/policyrep/terule.pxi (hunk 1 above)
Comment (tweksteen) [reply setools/policyrep/terule.pxi]: Done

File: setools/policyrep/terule.pxi [hunk 2]
```diff
@@ -267,12 +267,20 @@ cdef class IoctlSet(frozenset):
             sorted(self), key=lambda k, c=itertools.count(): k - next(c)))
 
 
+cdef class IoctlSet(XpermSet):
+
+    def __init__(self):
+        log = logging.getLogger(__name__)
+        log.warning("IoctlSet is deprecated, use XpermSet instead.")
+        super().__init__()
```
[the remark this answers is not in the document]
Comment (tweksteen) [reply setools/policyrep/terule.pxi]: Done

**Review by pebenito — APPROVED** [review]

**Comment by pebenito** [comment]
thanks!

**Pull request merged**